src/components/NewDeckForm/NewDeckForm.js

```diff
@@ -0,0 +1,158 @@
+import React, {useState} from "react";
+import {Form, Input, Button, Alert, Checkbox, Select} from "antd";
+import {useDispatch} from "react-redux";
+import {deckTags} from "../../utils/deckTags";
+import {createDeck} from "../../state/actions/decks";
+import styles from "../../components/RecentDecks/RecentDecks.module.css";
+
+const NewDeckForm = props => {
+  const [formValues, setFormValues] = useState({
+    name: "",
+    tags: [],
+    isPublic: false,
+  });
+
+  const dispatch = useDispatch();
+
+  const [state, setState] = useState({
+    success: null,
+  });
+
+  const handleSubmit = e => {
+    e.preventDefault();
+    const numberTags = formValues.tags.map(x => parseInt(x));
+    if (formValues.isPublic === true) {
+      dispatch(createDeck({...formValues, tags: numberTags, isPublic: "1"}))
+        .then(res => {
+          setState({
+            success: true,
+          });
+          setFormValues({
+            name: "",
+            tags: [],
+            isPublic: false,
+          });
+        })
+        .catch(err => {
+          setState({
+            success: false,
+          });
+        });
+    } else if (formValues.isPublic === false) {
+      dispatch(createDeck({...formValues, tags: numberTags, isPublic: "0"}))
+        .then(res => {
+          setState({
+            success: true,
+          });
+          setFormValues({
+            name: "",
+            tags: [],
+            isPublic: false,
+          });
+        })
+        .catch(err => {
+          setState({
+            success: false,
+          });
+        });
+    }
+  };
```

Comment:
Ideally you would push user to card creation after submit, something to do after card creation works

Reply:
We can do this when card creation works